packages/@aws-cdk/aws-stepfunctions-tasks/lib/athena/start-query-execution.ts

```diff
@@ -0,0 +1,289 @@
+          cdk.Stack.of(this).formatArn({
+            service: 'athena',
+            resource: 'workgroup',
+            resourceName: this.props.workGroup ? this.props.workGroup : 'primary',
+          }),
+
+        ],
+        actions: ['athena:getDataCatalog', 'athena:startQueryExecution'],
+      }),
+    ];
+
+    policyStatements.push(
+      new iam.PolicyStatement({
+        actions: ['s3:AbortMultipartUpload',
+          's3:CreateBucket',
+          's3:GetBucketLocation',
+          's3:GetObject',
+          's3:ListBucket',
+          's3:ListBucketMultipartUploads',
+          's3:ListMultipartUploadParts',
+          's3:PutObject'],
+        resources: [this.props.resultConfiguration?.outputLocation ? this.props.resultConfiguration?.outputLocation : '*'], // Need S3 location where data is stored https://docs.aws.amazon.com/athena/latest/ug/security-iam-athena.html
+      }),
+    );
+
+    policyStatements.push(
+      new iam.PolicyStatement({
+        actions: ['lakeformation:GetDataAccess'],
+        resources: [this.props.resultConfiguration?.outputLocation ? this.props.resultConfiguration?.outputLocation : '*'], // Workflow role permissions https://docs.aws.amazon.com/lake-formation/latest/dg/permissions-reference.html
+      }),
+    );
+
+    policyStatements.push(
+      new iam.PolicyStatement({
+        actions: ['glue:BatchCreatePartition',
+          'glue:BatchDeletePartition',
+          'glue:BatchDeleteTable',
+          'glue:BatchGetPartition',
+          'glue:CreateDatabase',
+          'glue:CreatePartition',
+          'glue:CreateTable',
+          'glue:DeleteDatabase',
+          'glue:DeletePartition',
+          'glue:DeleteTable',
+          'glue:GetDatabase',
+          'glue:GetDatabases',
+          'glue:GetPartition',
+          'glue:GetPartitions',
+          'glue:GetTable',
+          'glue:GetTables',
+          'glue:UpdateDatabase',
+          'glue:UpdatePartition',
+          'glue:UpdateTable'],
+        resources: [
+          cdk.Stack.of(this).formatArn({
+            service: 'glue',
+            resource: 'catalog',
+          }),
+          cdk.Stack.of(this).formatArn({
+            service: 'glue',
+            resource: 'database',
+            resourceName: this.props.queryExecutionContext?.databaseName ? this.props.queryExecutionContext?.databaseName : 'default',
+          }),
+          cdk.Stack.of(this).formatArn({
+            service: 'glue',
+            resource: 'table',
+            resourceName: '*', // access needed to query all tables https://docs.aws.amazon.com/athena/latest/ug/security-iam-athena.html
+          }),
+          cdk.Stack.of(this).formatArn({
+            service: 'glue',
+            resource: 'userdefinedfunction',
+            resourceName: '*', // users need access to start user defined functions from step functions https://docs.aws.amazon.com/athena/latest/ug/security-iam-athena.html
+          }),
+        ],
+      }),
+    );
```

Comment:
im unsure why permissions to `glue` api's are needed - the docs link leads to a list of topics but can you point to the specific ones that indicate that these policies are needed?